packages/xchain-radix/src/client.ts

```diff
@@ -0,0 +1,852 @@
+  Message,
+  NetworkId,
+  NotarizedTransaction,
+  PrivateKey,
+  PublicKey,
+  RadixEngineToolkit,
+  TransactionBuilder,
+  TransactionHash,
+  TransactionManifest,
+  address,
+  bucket,
+  decimal,
+  enumeration,
+  generateRandomNonce,
+} from '@radixdlt/radix-engine-toolkit'
+import {
+  AssetInfo,
+  Balance,
+  BaseXChainClient,
+  FeeType,
+  Fees,
+  Network,
+  PreparedTx,
+  Tx,
+  TxFrom,
+  TxHistoryParams,
+  TxParams,
+  TxTo,
+  TxType,
+  TxsPage,
+  XChainClientParams,
+  singleFee,
+} from '@xchainjs/xchain-client'
+import { getSeed } from '@xchainjs/xchain-crypto'
+import { Address, Asset, baseAmount } from '@xchainjs/xchain-util'
+import { bech32m } from 'bech32'
+import BIP32Factory, { BIP32Interface } from 'bip32'
+import { derivePath } from 'ed25519-hd-key'
+import * as ecc from 'tiny-secp256k1'
+// eslint-disable-next-line ordered-imports/ordered-imports
+import {
+  RadixChain,
+  XRD_DECIMAL,
+  assets,
+  bech32Lengths,
+  bech32Networks,
+  feesEstimationPublicKeys,
+  xrdRootDerivationPaths,
+} from './const'
+
+const xChainJsNetworkToRadixNetworkId = (network: Network): number => {
+  switch (network) {
+    case Network.Mainnet:
+    case Network.Stagenet:
+      return NetworkId.Mainnet
+    case Network.Testnet:
+      return NetworkId.Stokenet
+  }
+}
+
+type PartialTransactionPreviewResponse = {
+  receipt: {
+    status: 'Succeeded' | 'Failed' | 'Rejected'
+    fee_summary: {
+      execution_cost_units_consumed: number
+      finalization_cost_units_consumed: number
+      xrd_total_execution_cost: string
+      xrd_total_finalization_cost: string
+      xrd_total_royalty_cost: string
+      xrd_total_storage_cost: string
+      xrd_total_tipping_cost: string
+    }
+  }
+}
+
+/**
+ * The main client for the Radix network which is then wrapped by the {Client} adapting it to have
+ * a {BaseXChainClient} interface.
+ */
+export class RadixSpecificClient {
+  private innerNetwork: number
+  private innerGatewayClient: GatewayApiClient
+
+  constructor(networkId: number) {
+    this.innerNetwork = networkId
+    this.innerGatewayClient = RadixSpecificClient.createGatewayClient(networkId)
+  }
+
+  // #region Getters & Setters
+  public set networkId(networkId: number) {
+    this.innerNetwork = networkId
+    this.innerGatewayClient = RadixSpecificClient.createGatewayClient(networkId)
+  }
+
+  public get networkId(): number {
+    return this.innerNetwork
+  }
+
+  public get gatewayClient(): GatewayApiClient {
+    return this.innerGatewayClient
+  }
+  // #endregion
+
+  // #region Public Methods
+  public async currentEpoch(): Promise<number> {
+    return this.innerGatewayClient.status.getCurrent().then((status) => status.ledger_state.epoch)
+  }
+
+  public async currentStateVersion(): Promise<number> {
+    return this.innerGatewayClient.status.getCurrent().then((status) => status.ledger_state.state_version)
+  }
+
+  public async fetchBalances(address: string): Promise<Balance[]> {
```

Comment:
Really interesting feature the non fungible balances. As the other clients do not support it, and to be consistent with them, I would split the code creating two different functions, one for fungible balances which can be retrieve from the getBalance function and another one for the non fungible part

Reply:
@0xp3gasus I can totally split it but what would you like the getBalance function to return? just fungible tokens balances? that would be sort of inaccurate as non fungible tokens do contribute to the account balance

Reply:
EVM chains has also NFTs, but only ERC20 balances are returned. I suggest this change to be consistent with them and create a specific feature for NFTs for those clients which are compatible with them, for example, `getNFTBalances`

Reply:
fair enough. I'm splitting this